x-pack/legacy/plugins/monitoring/public/views/no_data/controller.js

```diff
@@ -48,79 +92,25 @@ export class NoDataController {
     };
   }
 
-  /*
-   * Start the page logic of observing scope changes, and changes to the data model.
-   * @param {Object} $scope Angular $scope of the view controller
-   * @param {Object} $injector Angular $injector service
-   * @param {Object} $executor Kibana $executor service for Angular
-   */
-  start($scope, $injector, $executor) {
-    const model = this;
-
-    const $http = $injector.get('$http');
-    const kbnUrl = $injector.get('kbnUrl');
-    const monitoringClusters = $injector.get('monitoringClusters');
-
-    const { updateModel } = new ModelUpdater($scope, model);
-    const checkers = [
-      new ClusterSettingsChecker($http),
-      new NodeSettingsChecker($http)
-    ];
-    const enabler = new Enabler($http, updateModel);
-
-    $scope.$$postDigest(() => {
-      startChecks(checkers, updateModel); // Start the checkers that use APIs for finding the reason for no data
-    });
-
-    $scope.$watch(
-      () => model,
-      props => {
-        render(
-          <I18nContext>
-            <NoData {...props} enabler={enabler} />
-          </I18nContext>,
-          document.getElementById(REACT_NODE_ID_NO_DATA)
-        );
-      },
-      true // deep watch
-    );
-
-    $scope.$watch(
-      () => model.hasData,
-      hasData => {
-        if (hasData) {
-          kbnUrl.redirect('/home'); // redirect if to cluster overview if data is found from background refreshes
-        }
-      }
+  render(enabler) {
+    const props = this;
+    /**
+     * Need to do this because {data} can either be an array or string, because of
+     * some race condtions.
```

Comment:
I'm not sure I'm following this? I don't quite understand what race condition is involved here. Plus, I'm not sure I see how it even matters based on what code you have inside the `if`. Do you not want to do that if the `props.data` is an array? Why?

Reply:
I guess it wasn't a race condition, but `props` and `props.reason` were spread as one object eg: `{ ...reason, ...props }` which overrides `reason.data`(string) property with `props.data`(array)